### PR TITLE
Support cosmetic filterlists

### PR DESCRIPTION
**Pull request by muodov**
This PR adds the optional support for cosmetic filterlist rules. Note that they are only used as a fallback, when autoconsent rules are not available for the given site. Autoconsent rules are still preferred as they can actively click reject buttons.

- Currently, we are using Easylist Cookie list, and it's bundled into the source code for performance reasons. Building with another filterlist is fairly simple, but we don't provide an official API for this at the moment.
- The test browser extension bundles the filterlist by default. To use the filterlist in the library, you need to use the correct import path (see below), and set `enableFilterList` config to `true`.

**Breaking changes**:
- the exports of the npm package are now limited to the following endpoints:
  - `import AutoConsent from '@duckduckgo/autoconsent'` will import the main library class. This was the main use-case before, so most likely your code will not break.
  - `import AutoConsent from '@duckduckgo/autoconsent/extra'` will import the same class with a bundled filterlist. While it will increase the site coverage, it will also significantly increase the bundle size.
  - `import rules from '@duckduckgo/autoconsent/rules/rules.json'` will import the JSON rules. This is added mainly for backward compatibility.
- All exports support CommonJS and ESM, so you can use `require()` too.
- If you need to import another source file directly, you can still do it by the full path, e.g. `import DomActions from './node_modules/@duckduckgo/autoconsent/lib/dom-actions.ts'`

### CODE REVIEW

File: lib/web.ts [hunk 1]
```diff
@@ -436,6 +479,91 @@ export default class AutoConsent {
...
+
+  filterListFallback() {
+    if (!this.filtersEngine) {
+      this.updateState({ lifecycle: 'nothingDetected' });
+      return false;
+    }
+
+    const cosmeticStyles = getCosmeticStylesheet(this.filtersEngine);
+
+    // this may be a false positive: sometimes filters hide unrelated elements that are not cookie pop-ups
```
Comment (muodov) [inline lib/web.ts:522]: note that we still need to do this check to know if we should report to the background script

File: lib/web.ts [hunk 2]
```diff
@@ -107,12 +139,17 @@ export default class AutoConsent {
...
+      Object.keys(declarativeRules.consentomatic).forEach((name) => {
+        this.addConsentomaticCMP(name, declarativeRules.consentomatic[name]);
+      });
+    }
+
+    if (declarativeRules.autoconsent) {
+      declarativeRules.autoconsent.forEach((ruleset) => {
+        this.addDeclarativeCMP(ruleset);
+      });
+    }
```
Comment (muodov) [inline lib/web.ts:152]: just a cleanup

File: lib/utils.ts [hunk 3]
```diff
@@ -79,6 +83,7 @@ export function normalizeConfig(providedConfig: any): Config {
     detectRetries: 20,
     isMainWorld: false,
     prehideTimeout: 2000,
+    enableFilterList: false,
```
Comment (muodov) [inline lib/utils.ts:86]: filterlist should be explicitly enabled by the browser

File: rules/filterlists/overrides.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+@@||duckduckgo.com^$document
+@@||duckduckgo.com^$elemhide
+! domain-level exceptions above are not working with the adblocker library
+duckduckgo.com#@#li[data-layout="ad"]
```
Comment (sammacbeth) [inline rules/filterlists/overrides.txt:4]: This one can be removed now, right?

**Review by sammacbeth — CHANGES_REQUESTED** [review]
Looks good - just a couple of minor changes I think would improve it.

File: package.json [hunk 5]
```diff
@@ -65,6 +76,8 @@
     "onlyPublishWithReleaseLabel": true
   },
   "dependencies": {
-    "tldts-experimental": "^6.1.37"
+    "@cliqz/adblocker": "^1.33.1",
+    "@cliqz/adblocker-content": "^1.33.1",
```
[the remark this answers is not in the document]
Comment (muodov) [reply package.json]: thanks for flagging, I updated it

File: lib/types.ts [hunk 6]
```diff
@@ -37,8 +37,8 @@ export interface DomActionsProvider {
 }
 
 export type RuleBundle = {
-  autoconsent: AutoConsentCMPRule[];
-  consentomatic: { [name: string]: ConsentOMaticConfig };
+  autoconsent?: AutoConsentCMPRule[];
```
[the remark this answers is not in the document]
Comment (muodov) [reply lib/types.ts]: you're right, I made it required

File: package.json [hunk 7]
```diff
@@ -2,29 +2,40 @@
...
     "test:lib": "web-test-runner",
     "test:sample": "playwright test tests/_sample-test.spec.ts --project webkit",
-    "fetch-fanboy-list": "wget https://www.fanboy.co.nz/fanboy-cookiemonster.txt",
-    "fetch-site-list": "curl https://s3.amazonaws.com/cdn.cliqz.com/re-consent/supported_sites.txt > sites.txt",
+    "fetch-easylist": "rm -f rules/filterlists/easylist_cookie_* && wget --directory-prefix=rules/filterlists/ https://raw.githubusercontent.com/easylist/easylist/refs/heads/master/easylist_cookie/easylist_cookie_general_hide.txt https://raw.githubusercontent.com/easylist/easylist/refs/heads/master/easylist_cookie/easylist_cookie_specific_hide.txt https://raw.githubusercontent.com/easylist/easylist/refs/heads/master/easylist_cookie/easylist_cookie_specific_uBO.txt https://raw.githubusercontent.com/easylist/easylist/refs/heads/master/easylist_cookie/easylist_cookie_international_specific_hide.txt https://raw.githubusercontent.com/easylist/easylist/refs/heads/master/easylist_cookie/easylist_cookie_allowlist_general_hide.txt https://raw.githubusercontent.com/easylist/easylist/refs/heads/master/easylist_cookie/easylist_cookie_allowlist.txt",
```
[the remark this answers is not in the document]
Comment (muodov) [reply package.json]: made changes as discussed: we're now tracking the resulting filterlist in git

File: rules/filterlists/overrides.txt (hunk 4 above)
Comment (muodov) [reply rules/filterlists/overrides.txt]: yep

**Pull request merged**